src/components/PopupWithForm.js

```diff
@@ -7,9 +7,11 @@ export default class PopupWithForm extends Popup {
     this._popupForm = this._popupElement.querySelector(".modal__form");
     this._handleFormSubmit = handleFormSubmit;
     this._handleSubmit = this._handleSubmit.bind(this);
+    this._popupButton = this._popupElement.querySelector(".modal__button");
   }
 
   _handleSubmit(evt) {
+    this._popupButton.textContent = "Saving...";
     evt.preventDefault();
     const formData = this._getInputValues();
     this._handleFormSubmit(formData);
```

Comment:
```js

    this.close();
    this._popupForm.reset();
```

They should be in `then`


You need to **reset** the form  in block `then` not to do that immediately after `submit` because when there is an error the popup will still be opened and the user can try to send the data again to the server without having to type the data into the inputs once more

Reply:
Updated this code for improved error handling. Adding new functionality to display for any errors.